pkg/utils/utils.go

```diff
@@ -169,6 +176,28 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 		if err != nil {
 			glog.Warningf("configSriovDevice(): unable to parse VFs for device %+v %q", iface.PciAddress, err)
 		}
+		// Set Sriov mode for ETH devices
+		if iface.SriovMode != "" && ifaceStatus.Name != "" {
+			link, err := netlink.LinkByName(ifaceStatus.Name)
+			if err != nil {
+				return err
+			}
+			if link.Attrs().EncapType == "ether" {
```

Comment:
do we really need to check the link state?
switchdev/legacy configuration is being done by devlink which is not aware of the link type

Reply:
yes, IB doesn't support changing sriov mode